agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/BaseDecorator.java

```diff
@@ -97,20 +93,39 @@ public Span onPeerConnection(final Span span, final InetSocketAddress remoteConn
 
   public Span onPeerConnection(final Span span, final InetAddress remoteAddress) {
     assert span != null;
-    String hostName = remoteAddress.getHostName();
-    if (!hostName.equals(remoteAddress.getHostAddress())) {
-      span.setAttribute(SemanticAttributes.NET_PEER_NAME.key(), remoteAddress.getHostName());
-    }
-    span.setAttribute(SemanticAttributes.NET_PEER_IP.key(), remoteAddress.getHostAddress());
+    String peerName = remoteAddress.getHostName();
+    String peerIp = remoteAddress.getHostAddress();
+    setPeer(span, peerName, peerIp);
+    return span;
+  }
 
-    String peerService = mapToPeer(hostName);
-    if (peerService == null) {
-      peerService = mapToPeer(remoteAddress.getHostAddress());
-    }
+  private static void setPeer(final Span span, String peerName) {
+    assert span != null;
+    span.setAttribute(SemanticAttributes.NET_PEER_NAME.key(), peerName);
+    String peerService = mapToPeer(peerName);
     if (peerService != null) {
       span.setAttribute("peer.service", peerService);
     }
-    return span;
+  }
```

Comment:
i think remove this method and call `setPeer(span, peerName, null)` directly, and i think restructure setPeer below a bit to not to delegate to this method